src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -133,7 +133,7 @@ CHIP_ERROR UDPEndPointImplLwIP::LwIPBindInterface(struct udp_pcb * aUDP, Interfa
 InterfaceId UDPEndPointImplLwIP::GetBoundInterface() const
 {
 #if HAVE_LWIP_UDP_BIND_NETIF
-    return InterfaceId(netif_get_by_index(mUDP->netif_idx));
+    return InterfaceId(netif_get_by_index(mUDP->netif_idx + 1));
```

Comment:
Why "+ 1"?

The intent of the calling code is to do "all interfaces".  Why is adding 1 here the right thing?

Reply:
`mUDP->netif_idx` is passing a value of `0`, `netif_get_by_index` returns NULL in that case. Since there is a check for `0`, ideally the right thing would be for the netif function to return the interface at 0 position.